content/blog/2025-03-27-valkey-8-1-0-rc1.md

```diff
@@ -0,0 +1,282 @@
++++
+# `title` is how your post will be listed and what will appear at the top of the post
+title= "Valkey 8.1: Continuing to Deliver Enhanced Performance and Reliability"
+# `date` is when your post will be published.
+# For the most part, you can leave this as the day you _started_ the post.
+# The maintainers will update this value before publishing
+# The time is generally irrelvant in how Valkey published, so '01:01:01' is a good placeholder
+date= 2025-04-02 01:01:01
+# 'description' is what is shown as a snippet/summary in various contexts.
+# You can make this the first few lines of the post or (better) a hook for readers.
+# Aim for 2 short sentences.
+description= "Valkey 8.1 is now generally available! Come learn about the exciting improvements in performance, reliability, and observability that are available in this new version."
+# 'authors' are the folks who wrote or contributed to the post.
+# Each author corresponds to a biography file (more info later in this document)
+authors= [ "rdias", "mvisser" ]
++++
+
+
+The Valkey community is proud to unveil the new release of Valkey 8.1,
+a minor version update designed to further enhance performance, reliability, observability and usability
+over Valkey 8.0 for all Valkey installations. 
+In this blog, we'll dive a bit deeper into some of these areas 
+and talk about the exciting features built for this release.
+
+## Performance
+
+There are many changes in 8.1 that improve Valkey's performance either by decreasing the latency of some operations, increasing the throughput of some operations and reducing the memory footprint of the overall system.
+
+
+### The New Hashtable
+
+One of the main changes responsible for several performance improvements is the new hashtable implementation that is used as the main key-value store in Valkey and also used in the Hash, Set, and Sorted Set data types implementation.
+
+Valkey is a distributed key-value and therefore is no surprise that a hashtable data structure performs an important role in the way Valkey works.
+
+Almost all commands go through the main key-value store where the several datatypes values are stored, and therefore the hashtable implementation is present in the critical path of most workloads.
+
+The new hashtable implementation is a complete rewrite of the previous hashtable implementation, which reduces the memory usage by roughly 20 bytes per key-value pair. The new implementation also helps improve the Valkey server supported throughput by roughly 10% compared to 8.0 version for pipeline workloads when I/O threading is not used.
```

Comment:
> roughly 10% compared to 8.0 version for pipeline workloads when I/O threading is not used.

can we add a link to the issue or PR where 10% is reported, for more context?

Reply:
I got this locally (laptop, loopback interface, benchmark and server on the same machine) but it wasn't consistently reproduced in proper benchmarks, so I didn't claim this in the bash table blog post. We should probably leave this out.

Btw, the hash table blog post is live so you can link to it.